go/tasks/plugins/array/core/state.go

```diff
@@ -238,11 +239,18 @@ func SummaryToPhase(ctx context.Context, minSuccesses int64, summary arraystatus
 				// TODO: preferable to auto-combine to array tasks for now.
 				totalFailures += count
 			}
+		} else if phase.IsWaitingForResources() {
+			totalWaitingForResources += count
 		} else {
 			totalRunning += count
 		}
 	}
 
+	if totalWaitingForResources > 0 {
```

Comment:
That is correct. I updated the existing function to check for `totalRunning+totalSuccesses+totalWaitingForResources < minSuccesses` instead of just `totalRunning+totalSuccesses < minSuccesses`